### PR TITLE
sUSD Metapool

### PR DESCRIPTION
**Pull request by hammeiam**
- Incorporate the concept of a metapool across the app
- Add the susd metapool
- Refactor useContract to be slimmer
- Refactor routes to be derived from our pools list

All flows tested and working locally

### CODE REVIEW

File: src/constants/index.ts [hunk 1]
```diff
@@ -238,9 +287,26 @@ export const D4_SWAP_TOKEN = new Token(
...
+const SUSD_CONTRACT_ADDRESSES: { [chainId in ChainId]: string } = {
+  [ChainId.MAINNET]: "0x57Ab1ec28D129707052df4dF418D58a2D46d5f51",
+  [ChainId.ROPSTEN]: "",
+  [ChainId.HARDHAT]: "0x0E801D84Fa97b50751Dbf25036d067dCf18858bF",
+}
+export const SUSD = new Token(
+  SUSD_CONTRACT_ADDRESSES,
+  18,
+  "sUSD",
+  "nusd",
```
Comment (alphastorm) [inline src/constants/index.ts:304]: s/nusd/susd

https://www.coingecko.com/en/coins/susd

File: src/hooks/usePoolData.ts [hunk 2]
```diff
@@ -114,12 +120,25 @@ export default function usePoolData(
       )
         return
       const POOL = POOLS_MAP[poolName]
+      const effectivePoolTokens = POOL.underlyingPoolTokens || POOL.poolTokens
+      const isMetaSwap = POOL.metaSwapAddresses != null
+      let metaSwapContact = null as MetaSwap | null
```
Comment (alphastorm) [inline src/hooks/usePoolData.ts:125]: s/Contact/Contract

**Review by alphastorm — COMMENTED** [review]
coverage CI is broken at the moment

File: src/constants/index.ts (hunk 1 above)
Comment (penandlim) [reply src/constants/index.ts:304]: The id required for API call is still nusd :S

**Comment by penandlim** [comment]
Looks good but is there a way we can let users deposit with the USDv2 LP token instead? 
Current setup would make it so the exsiting LPs need to withdraw from USDv2 then redeposit to the susd meta pool.

**Comment by penandlim** [comment]
image
Found an icon size issue - sUSD seems too big @sandrasong

**Comment by sandrasong** [comment]
@penandlim Fixed the different icon size bug. The reason was the icons were different size. We used 18x18 in the previous design. But icons in current design are 24 x 24. I will create a ticket to update all icons.

**Review by alphastorm — APPROVED** [review]
lgtm, tested locally and everything works 🥳 

we should also add the option to withdraw wrapped as well

**Review by penandlim — APPROVED** [review]

**Pull request merged**